resources/js/components.js

```diff
@@ -1,12 +1,18 @@
 (() => {
     const components = {
         // Eager load all components not containing an extra . in the name
-        ...import.meta.glob(['./components/*([^\.]).vue'], { eager: true, import: 'default' }),
+        ...import.meta.glob(['./components/**/*([^\.]).vue'], { eager: true, import: 'default' }),
         // Lazy load all components not ending with .lazy.vue
-        ...import.meta.glob(['./components/*.lazy.vue'], { eager: false, import: 'default' })
+        ...import.meta.glob(['./components/**/*.lazy.vue'], { eager: false, import: 'default' })
     };
     for (const path in components) {
+        let componentName = path
+            .split('/').pop() // Remove directories
+            .split('.').shift() // Remove extension
+            .replace(/[A-Z]/g, letter => `-${letter.toLowerCase()}`) // PascalCase to snake_case
+            .substr(1) // Remove the starting dash
```

Comment:
Transforming PascalCase shouldn't be necessary as Vue does this for you https://vuejs.org/guide/components/registration.html#component-name-casing

The `substr` could introduce unexpected behaviour when a component does not start with a capital letter.

Reply:
Maybe it should; but didn't

Reply:
If all else fails, a regex like `(^.|[A-Z])` would at least fix the substr issue by also always matching the first character.

Reply:
Nice! Thanks!